sdk/core/core-util/src/reshape.ts

```diff
@@ -0,0 +1,245 @@
+
+  if (rollingUpSelector) {
+    yield { type: "arrayIterator", remainingSelector: partialResult.join("") };
+  } else if (partialResult.length) {
+    yield { type: "propertyName", value: partialResult.join("") };
+  }
+}
+
+/**
+ * Options for configuring a selected property.
+ */
+export interface ReshapeOptions {
+  /**
+   * The new property name for the selected property.
+   */
+  newPropertyName?: string;
+  /**
+   * A function that maps the current property value to a new one.
+   */
+  mapFunction?: (value: unknown) => unknown;
+}
+
+/**
+ * A simple helper method that allows you to modify a JS object.
+ * The original argument is mutated in place, but also returned
+ * in order to make casting to the final type easier.
+ *
+ * Example usage:
+ * ```ts
+ * // changes input to look like `{b: 1}`
+ * reshape({a: 1}, "a", "b");
+ *
+ * ```
+ * @param object - The object to modify.
+ * @param selector - The selector for the property to modify.
+ * Separate property names with `.`,
+ * iterate over arrays with `[]`,
+ * and escape these characters with `\`.
+ * @param newPropertyName - The new property name for the selected property.
+ */
+export function reshape(object: unknown, selector: string, newPropertyName: string): unknown;
+/**
+ * A simple helper method that allows you to modify a JS object.
+ * The original argument is mutated in place, but also returned
+ * in order to make casting to the final type easier.
+ *
+ * Example usage:
+ * ```ts
+ * // changes input to look like `{a: "hello 1"}`
+ * reshape({a: 1}, "a", (value) => `hello ${value}`);
+ *
+ * ```
+ * @param object - The object to modify.
+ * @param selector - The selector for the property to modify.
+ * Separate property names with `.`,
+ * iterate over arrays with `[]`,
+ * and escape these characters with `\`.
+ * @param mapFunction - A function that maps the current property value to a new one.
+ */
+export function reshape(
+  object: unknown,
+  selector: string,
+  mapFunction: (value: unknown) => unknown
+): unknown;
+/**
+ * A simple helper method that allows you to modify a JS object.
+ * The original argument is mutated in place, but also returned
+ * in order to make casting to the final type easier.
+ *
+ * Example usage:
+ * ```ts
+ * // changes input to look like `{b: "hello 1"}`
+ * reshape({a: 1}, "a", "b", (value) => `hello ${value}`);
+ *
+ * ```
+ * @param object - The object to modify.
+ * @param selector - The selector for the property to modify.
+ * Separate property names with `.`,
+ * iterate over arrays with `[]`,
+ * and escape these characters with `\`.
+ * @param newPropertyName - The new property name for the selected property.
+ * @param mapFunction - A function that maps the current property value to a new one.
+ */
+export function reshape(
+  object: unknown,
+  selector: string,
+  newPropertyName: string,
+  mapFunction: (value: unknown) => unknown
+): unknown;
+/**
+ * A simple helper method that allows you to modify a JS object.
```

Comment:
Just want to clarify this goal for this helper is to aim at all different cases between RLC and Modular or just for common cases?

Reply:
Yeah, this definitely answered my question.